.buildkite/steps/documentation.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -ueo pipefail
+
+ln -nsf . terraform-provider-buildkite && cd terraform-provider-buildkite
+
+echo "--- :terraform: Running make docs"
+make docs
+
+echo "--- :git: Checking for changes"
+git diff --exit-code docs &>/dev/null && true
+
+docs_changes="$?"
+
+if [ "${docs_changes:-0}" -ne 0 ] ; then
+	echo "+++ :bk-status-failed: Documentation changes detected!!!"
+	git status --short
```

Comment:
I wonder should we output this as an annotation on the build to make it more front and centre? Non-blocking, just a general thought.

Reply:
Yeah thinking adding this as an annotation would make it better over all as things can be missed when within Job Logs